Modules/FIT/FT0/src/LaserAgingFT0Task.cxx

```diff
@@ -0,0 +1,176 @@
+  getObjectsManager()->setDefaultDrawOptions(mHistAmp2ADC0.get(), "COLZ");
+  getObjectsManager()->setDefaultDrawOptions(mHistAmp2ADC1.get(), "COLZ");
+
+  std::vector<unsigned int> vecAmpCut;
+  std::vector<unsigned int> vecChannelIDs;
+  std::vector<unsigned int> vecRefPMTChannelIDs;
+
+  if (auto param = mCustomParameters.find("cutAmpl"); param != mCustomParameters.end()) {
+    const auto ampCut = param->second;
+    const std::string del = ",";
+    vecAmpCut = parseParameters<unsigned int>(ampCut, del);
+  } else {
+    vecAmpCut.push_back(0);
+  }
+  if (auto param = mCustomParameters.find("ChannelIDs"); param != mCustomParameters.end()) {
+    const auto chIDs = param->second;
+    const std::string del = ",";
+    vecChannelIDs = parseParameters<unsigned int>(chIDs, del);
+  } else {
+    for (unsigned int iCh = 0; iCh < o2::ft0::Constants::sNCHANNELS_PM; iCh++)
+      vecChannelIDs.push_back(iCh);
+  }
+  if (auto param = mCustomParameters.find("RefPMTChannelIDs"); param != mCustomParameters.end()) {
+    const auto chIDs = param->second;
+    const std::string del = ",";
+    vecRefPMTChannelIDs = parseParameters<unsigned int>(chIDs, del);
+  }
+
+  for (const auto& entry : vecAmpCut) {
+    mSetAmpCut.insert(entry);
+  }
+  for (const auto& entry : vecChannelIDs) {
+    mSetAllowedChIDs.insert(entry);
+  }
+  for (const auto& entry : vecRefPMTChannelIDs) {
+    mSetRefPMTChIDs.insert(entry);
+  }
+
+  for (const auto& RefPMTChID : mSetRefPMTChIDs) {
+    auto pairHistAmpVsBCADC0 = mMapHistAmpVsBCADC0.insert({ RefPMTChID, new TH2F(Form("ADCzero_Amp_vs_BC_channel%i", RefPMTChID), Form("Amplitude vs BC, channel %i;Amp;BC", RefPMTChID), 1000, 0, 1000, 1000, 0, 1000) });
+    auto pairHistAmpVsBCADC1 = mMapHistAmpVsBCADC1.insert({ RefPMTChID, new TH2F(Form("ADCone_Amp_vs_BC_channel%i", RefPMTChID), Form("Amplitude vs BC, channel %i;Amp;BC", RefPMTChID), 1000, 0, 1000, 1000, 0, 1000) });
+    if (pairHistAmpVsBCADC0.second) {
+      getObjectsManager()->startPublishing(pairHistAmpVsBCADC0.first->second);
+    }
+    if (pairHistAmpVsBCADC1.second) {
+      getObjectsManager()->startPublishing(pairHistAmpVsBCADC1.first->second);
+    }
+  }
+}
+
+void LaserAgingFT0Task::startOfActivity(const Activity& activity)
+{
+  // THUS FUNCTION BODY IS AN EXAMPLE. PLEASE REMOVE EVERYTHING YOU DO NOT NEED.
+  ILOG(Debug, Devel) << "startOfActivity " << activity.mId << ENDM;
+  mHistAmp2ADC0->Reset();
+  mHistAmp2ADC1->Reset();
+  for (auto& entry : mMapHistAmpVsBCADC0)
+    entry.second->Reset();
+  for (auto& entry : mMapHistAmpVsBCADC1)
+    entry.second->Reset();
+}
+
+void LaserAgingFT0Task::startOfCycle()
+{
+  // THUS FUNCTION BODY IS AN EXAMPLE. PLEASE REMOVE EVERYTHING YOU DO NOT NEED.
+  ILOG(Debug, Devel) << "startOfCycle" << ENDM;
+}
+
+void LaserAgingFT0Task::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  unsigned int costumAmplCut = 0;
+  for (const auto& cutAmpl : mSetAmpCut) {
+    costumAmplCut = cutAmpl;
+  }
```

Comment:
I think some change is needed here, either the amp cut is channel wise or it is not?